src/workers/python/PythonWorker.worker.ts

```diff
@@ -20,58 +21,48 @@ declare function loadPyodide(args: { indexURL: string; fullStdLib: boolean }): P
  */
 class PythonWorker extends Backend {
     pyodide: Pyodide;
-    initialized: boolean;
 
     constructor() {
         super();
         this.pyodide = {} as Pyodide;
-        this.initialized = false;
     }
 
     private convert(data: any): any {
         let converted = data;
-        if ("toJs" in data) {
-            converted = data.toJs();
+        if (data.toJs) {
+            converted = data.toJs({ dict_converter: Object.fromEntries });
         }
-        return Object.fromEntries(converted);
+        if (converted instanceof Map) {
+            converted = Object.fromEntries(converted);
+        }
```

Comment:
In theory yes, but I was still getting Map-objects at some point. I'll take a second look.

Reply:
After some testing, this is still required. It seems like that if the top-object is a dictionary, you still end up with a Map instead of an object. I don't know if this is something we want to raise with Pyodide or if this is acceptable.